README.md

```diff
@@ -4,28 +4,149 @@ Mediawiki title normalizetion, that conforms to the normalization rules used in
+
+
+| Param | Type | Description |
+| --- | --- | --- |
+| id | <code>number</code> | The namespace identifier |
+| siteInfo | <code>[SiteInfo](#SiteInfo)</code> | The site metadata information. |
+
+<a name="Namespace+isSpecial"></a>
+### namespace.isSpecial() ⇒ <code>boolean</code>
+Checks whether namespace is `Special`.
+
+**Kind**: instance method of <code>[Namespace](#Namespace)</code>  
+<a name="Namespace+isMain"></a>
+### namespace.isMain() ⇒ <code>boolean</code>
+Checks whether namespace is `Main`.
+
+**Kind**: instance method of <code>[Namespace](#Namespace)</code>  
+<a name="Namespace+isTalk"></a>
+### namespace.isTalk() ⇒ <code>boolean</code>
+Checks whether namespace is `Talk`.
+
+**Kind**: instance method of <code>[Namespace](#Namespace)</code>  
+<a name="Namespace+isUser"></a>
+### namespace.isUser() ⇒ <code>boolean</code>
+Checks whether namespace is `User`.
+
+**Kind**: instance method of <code>[Namespace](#Namespace)</code>  
+<a name="Namespace+isUserTalk"></a>
+### namespace.isUserTalk() ⇒ <code>boolean</code>
+Checks whether namespace is `User_Talk`.
+
+**Kind**: instance method of <code>[Namespace](#Namespace)</code>  
+<a name="Namespace+getNormalizedText"></a>
+### namespace.getNormalizedText() ⇒ <code>string</code>
+Get the canonical name string for this namespace.
+
+**Kind**: instance method of <code>[Namespace](#Namespace)</code>  
+<a name="Namespace+getCase"></a>
+### namespace.getCase() ⇒ <code>string</code>
+Returns the case parameter for the namespace
+
+**Kind**: instance method of <code>[Namespace](#Namespace)</code>  
+<a name="Namespace.fromText"></a>
+### Namespace.fromText(text, siteInfo) ⇒ <code>[Namespace](#Namespace)</code> &#124; <code>undefined</code>
+Creates a namespace instance from namespace text or a namespace alias
+
+**Kind**: static method of <code>[Namespace](#Namespace)</code>  
+**Returns**: <code>[Namespace](#Namespace)</code> &#124; <code>undefined</code> - a namespace or undefined if it wasn't found.  
+
+| Param | Type | Description |
+| --- | --- | --- |
+| text | <code>string</code> | Namespace name text. |
+| siteInfo | <code>[SiteInfo](#SiteInfo)</code> | the site information. |
+
+<a name="Namespace.main"></a>
+### Namespace.main(siteInfo) ⇒ <code>[Namespace](#Namespace)</code>
+Creates a namespace object for a `Main` namespace.
+
+**Kind**: static method of <code>[Namespace](#Namespace)</code>  
+
+| Param | Type | Description |
+| --- | --- | --- |
+| siteInfo | <code>[SiteInfo](#SiteInfo)</code> | the site information. |
+
+<a name="Title"></a>
+## Title
+**Kind**: global class  
+
+* [Title](#Title)
+    * [new Title(key, namespace, siteInfo, [fragment])](#new_Title_new)
+    * _instance_
+        * [.getNormalizedText()](#Title+getNormalizedText) ⇒ <code>string</code>
+        * [.getNamespace()](#Title+getNamespace) ⇒ <code>[Namespace](#Namespace)</code>
+    * _static_
+        * [.fromPrefixedText(title, siteInfo)](#Title.fromPrefixedText) ⇒ <code>[Title](#Title)</code>
```

Comment:
The MW implementation calls this `newFromText`. This name seems more consistent with `getPrefixedText()` though, which returns the normalized title with namespace prefix, but with spaces instead of underscores (as opposed to `getPrefixedDBKey`).